edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Edgelet/ModuleManagementHttpClient.cs

```diff
@@ -17,6 +17,10 @@ public class ModuleManagementHttpClient : IModuleManager, IIdentityManager, IDev
     {
         readonly ModuleManagementHttpClientVersioned inner;
 
+        readonly TimeSpan clientTicketTimeout = TimeSpan.FromSeconds(240);
```

Comment:
Just curious, how did you pick this value?

Reply:
I know this isn't the final, proper fix. But if the communication chain is agent -> aziot-edged -> aziot-identityd -> iothub, I fear what will happen if this timeout is reached is that the in progress agent -> aziot-edged call will be cancelled and identityd will still be trying to communicate with hub. This causes more requests to be in the system, and some of these requests are not going to end up returning a module identity, necessitating more calls in the future.